### PR TITLE
Fixes Restart of Learners by Serializing Optimizers

### PR DESCRIPTION
**Pull request by webepasc**

### CODE REVIEW

File: source/modules/module.cpp [hunk 1]
```diff
@@ -58,6 +58,10 @@
 #include "solver/integrator/montecarlo/MonteCarlo.hpp"
 #include "solver/integrator/quadrature/Quadrature.hpp"
 #include "solver/deepSupervisor/deepSupervisor.hpp"
+#include "solver/deepSupervisor/optimizers/fAdam/fAdam.hpp"
```
Comment (wadaniel) [inline source/modules/module.cpp:61]: hmm lets discuss this, its confusing to offer two different versions of Adam, maybe we will keep only one? the f version and name it Adam?

Comment (wadaniel) [reply source/modules/module.cpp:61]: the fast version is now the same as the normal, except that it is written with floats, there are no more diffs?

Comment (webepasc) [reply source/modules/module.cpp:61]: Yes, the main difference regarding the implementation is that the parameters are floats and not double. However, there is more: For the optimization problem, the user provides a computational model. This model is evaluated via a sample that goes through the conduit. The float optimizers are more lightweight and do not rely on this process and therefore do not implement the runGeneration function.  I think this specialization for neural networks makes sense as a "pure" optimization problem will typically be in double (the speedup for float is only relevant for neural networks).

Comment (wadaniel) [reply source/modules/module.cpp:61]: ok

File: examples/supervised.learning/sineWave/plot-loss.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import os
```
Comment (wadaniel) [inline examples/supervised.learning/sineWave/plot-loss.py:1]: can we include this file when running tests?

Comment (webepasc) [reply examples/supervised.learning/sineWave/plot-loss.py:2]: done

File: examples/supervised.learning/mnist/run-mnist-autoencoder.py [hunk 3]
```diff
@@ -56,7 +57,7 @@
 
 ### Using a neural network solver (deep learning) for inference
 
-e["Solver"]["Termination Criteria"]["Max Generations"] = 1
+e["Solver"]["Termination Criteria"]["Max Generations"] = 0
```
Comment (wadaniel) [inline examples/supervised.learning/mnist/run-mnist-autoencoder.py:60]: 0 Generations? Does it work?

**Review by wadaniel — APPROVED** [review]

File: examples/supervised.learning/mnist/run-mnist-autoencoder.py (hunk 3 above)
Comment (webepasc) [reply examples/supervised.learning/mnist/run-mnist-autoencoder.py:60]: Yes, if you see further down in the loop, the counter is increased to 1. Before the first iteration did two generations..

**Pull request merged**